### DISCUSSION TITLE
Incremental appearing of images similar to incremental lists

### DISCUSSION THREAD

**Discussion by jweickm** [marp-team/marp#289] [Ideas] [answered]
It would be amazing if there were a way to display the images on the slides incrementally in a similar way that we do with incremental lists. Currently, the only way to do this (that I know of), is to put them in an incremental list like 

```
* ![](image1)
* ![](image2)
```

However, this leaves some unsightly bullets in front of the images. 
It would be amazing if we could just ignore bullets when in front of an image tag or have another syntax like 
```
![](normal_img)
?[](incremental_img)
```

Is this something that would be interesting for other users as well?

**Comment by yhatt** [comment, marked as answer]
Marp has a rooted policy that _the additional syntax must prevent breaking CommonMark syntax and keep compatibility with general Markdown documents_. `?[]()` syntax would not be recognized as the image syntax in the other Markdown tools in common, so we have no plans to adopt it as the incremental image. (But we have extensible architecture based on [markdown-it plugin](https://www.npmjs.com/search?q=keywords:markdown-it-plugin) ecosystem for extending the syntax as you like 😃)

[Using custom style through `<style>`](https://marpit.marp.app/theme-css?id=tweak-style-through-markdown) to remove bullets from the fragmented list is [a valid way as CommonMark](https://spec.commonmark.org/0.30/#example-172). Refer to https://github.com/marp-team/marp/discussions/270#discussioncomment-2209927.

> ```
> # Hidden bullets
> 
> <style scoped>li { list-style-type: none}</style>
> 
> * These bullets
> * are hidden
> * to the viewer
> ```
>
> _&mdash; https://github.com/marp-team/marp/discussions/270#discussioncomment-2209927_

**Comment by jweickm** [comment]
Hi yhatt, 

お返事いただきありがとうございます。いつもお世話になっております。

I understand that the downside of introducing new syntax elements outweighs their potential benefits. I also had a blast reading through the thread that you linked, referring also to best practices for presentations, and the like. I use Marpit mainly for academic presentations in a teaching context, so I understand that this is not the main focus of the Marpit framework. However, I also want to say that it is still incredibly time-saving and and a great joy to be able to create beautiful + consistent presentations just with the keyboard.

+1 for the solution that you provided. This shall suffice for my application! Thanks again! 😊

Best, 
jweickm
